cachi2/core/package_managers/yarn/project.py

```diff
@@ -31,43 +39,56 @@ def cache_path(self) -> str:
 
         Fallback to the default path in case the configuration key is missing.
         """
-        return NotImplemented
+        return self._data.get("cacheFolder", "./.yarn/cache")
 
     @property
     def registry_server(self) -> str:
         """Get the globally configured registry server.
 
         Fallback to the default server in case the configuration key is missing.
         """
-        return NotImplemented
+        return self._data.get("npmRegistryServer", "https://registry.yarnpkg.com")
 
     @property
     def yarn_path(self) -> Optional[str]:
         """Path to the yarn script present in this directory."""
-        return NotImplemented
+        try:
+            return self._data.get("yarnPath")
+        except KeyError:
+            log.warning(f"No yarnPath in {self._path}")
+            return None
 
     @property
     def yarn_version(self) -> Optional[str]:
         """Yarn version used in this project.
 
         Extracted from the contents of the yarnPath configuration, if it is present.
         """
-        return NotImplemented
+        try:
+            yarn_path = self._data.get("yarnPath")
+            return re.search(r"([0-9]+\.[0-9]+\.[0-9]+)", yarn_path).group(0)
```

Comment:
+1

rather than trying to match npm semver, how about we just take this part of the filename as version?
```
some/path/yarn-(.*).cjs
               ^^^^
```

Reply:
nice :+1:  changed the regex